journals/apps/journals/models.py

```diff
@@ -310,9 +311,10 @@ def data(self):
         Return the contents of the document as base64 encoded
         data used as input to elasticsearch ingest-attachment plugin
         '''
-        self.file.open()
-        contents = base64.b64encode(self.file.read()).decode('ascii')
-        self.file.close()
+        # converting to base64 is (Size * 8)/6 times bigger than binary file
+        # determine the max number of bytes we can read to not exceed the max upload size
+        read_max = (settings.MAX_ELASTICSEARCH_UPLOAD_SIZE * 6) / 8
+        contents = base64.b64encode(os.pread(self.file.fileno(), int(read_max), 0)).decode('ascii')
```

Comment:
I'm not sure if `os.pread` is going to cause any issues when we're using S3 in prod. `file.read()` can take a size though, so you can skip the middleman here. just do: 
`base64.b64encode(self.file.read(read_max)).decode('ascii')`